zerver/lib/actions.py

```diff
@@ -1183,11 +1199,14 @@ def check_send_message(sender, client, message_type_name, message_to,
                        subject_name, message_content, realm=None, forged=False,
                        forged_timestamp=None, forwarder_user_profile=None, local_id=None,
                        sender_queue_id=None):
-    # type: (UserProfile, Client, Text, Sequence[Text], Optional[Text], Text, Optional[Realm], bool, Optional[float], Optional[UserProfile], Optional[Text], Optional[Text]) -> int
+    # type: (UserProfile, Client, Text, Sequence[Text], Optional[Text], Text, Optional[Realm], bool, Optional[float], Optional[UserProfile], Optional[Text], Optional[Text]) -> Optional[int]
     message = check_message(sender, client, message_type_name, message_to,
                             subject_name, message_content, realm, forged, forged_timestamp,
                             forwarder_user_profile, local_id, sender_queue_id)
-    return do_send_messages([message])[0]
+    ids = do_send_messages([message])
+    if len(ids) > 0:
+        return ids[0]
+    return None
```

Comment:
Currently the return value of `check_send_message` is not being used except in tests and in `send_message_backend` view. The tests are not breaking with this change. Also when the view returns `None` as message id, the UI deletes the message, which helps our cause of dropping the trigger message.